servers/quarkus-server/build.gradle.kts

```diff
@@ -107,47 +107,49 @@ dependencies {
 
 preferJava11()
 
-val openApiSpecDir = project.buildDir.resolve("openapi-extra").relativeTo(project.projectDir)
-
-project.extra["quarkus.smallrye-openapi.store-schema-directory"] =
-  "${project.buildDir.relativeTo(project.projectDir)}/openapi"
-
-project.extra["quarkus.smallrye-openapi.additional-docs-directory"] = "$openApiSpecDir"
-
-project.extra["quarkus.package.type"] =
-  if (withUberJar()) "uber-jar" else if (project.hasProperty("native")) "native" else "fast-jar"
-
-val useDocker = project.hasProperty("docker")
-
 val pullOpenApiSpec by
   tasks.registering(Sync::class) {
     destinationDir = openApiSpecDir
     from(project.objects.property(Configuration::class).value(openapiSource))
   }
 
+val openApiSpecDir = buildDir.resolve("openapi-extra")
+val useDocker = project.hasProperty("docker")
+val packageType =
+  if (withUberJar()) "uber-jar" else if (project.hasProperty("native")) "native" else "fast-jar"
+val quarkusBuilderImage = libs.versions.quarkusBuilderImage.get()
+
+quarkus {
+  quarkusBuildProperties.put("quarkus.package.type", packageType)
+  quarkusBuildProperties.put(
+    "quarkus.smallrye-openapi.store-schema-directory",
+    buildDir.resolve("openapi").toString()
+  )
+  quarkusBuildProperties.put(
+    "quarkus.smallrye-openapi.additional-docs-directory",
+    openApiSpecDir.toString()
+  )
+  quarkusBuildProperties.put("quarkus.native.builder-image", quarkusBuilderImage)
+  if (useDocker) {
+    quarkusBuildProperties.put("quarkus.native.container-build", "true")
+    quarkusBuildProperties.put("quarkus.container-image.build", "true")
+  }
+}
+
 val quarkusBuild =
   tasks.named<QuarkusBuild>("quarkusBuild") {
-    outputs.cacheIf { false }
+    outputs.doNotCacheIf("Do not add huge cache artifacts to build cache") { true }
     dependsOn(pullOpenApiSpec)
-    inputs.property("quarkus.package.type", project.extra["quarkus.package.type"])
+    inputs.property("quarkus.package.type", packageType)
     inputs.property("final.name", quarkus.finalName())
     inputs.property("container-build", useDocker)
-    val quarkusBuilderImage = libs.versions.quarkusBuilderImage.get()
     inputs.property("builder-image", quarkusBuilderImage)
     if (useDocker) {
       // Use the "docker" profile to just build the Docker container image when the native image's
       // been built
       nativeArgs { "container-build" to true }
     }
     nativeArgs { "builder-image" to quarkusBuilderImage }
```

Comment:
Do we need this still? This comment was removed: `the nativeArgs{} thing above doesn't really work` :thinking: